spreadsheets.js

```diff
@@ -537,37 +541,109 @@ function update_result(service_account_auth, key, colname, result, callback){
             }
             var result_cell = row[colname];
 
-            //make the update to the cell
-            //parse the links out of the function
-            //update only the result - will create a function to update the gamelink
-            
-            if(reversed){
-                //switch the result
-               var lhs = result.split('-')[0];
-               var rhs = result.split('-')[1];
-               result = rhs + '-' + lhs;
+            //a gamelink was passed in and the names were reversed
+            if(gamelink_id && reversed){
+                //this is an error - games must be played by proper colors
+                return callback("the colors are reversed.", true);
+            }
+            if(result_string != "-"){
+                //this is just the addition of a result
+                //we will use the gamelink to verify the result if we have it
+                if(reversed){
+                    //switch the result
+                    var lhs = result_string.split('-')[0];
+                    var rhs = result_string.split('-')[1];
+                    result_string = rhs + '-' + lhs;
+                }
             }
 
-            var gamelink = ' ';            
-
+            //get the formula, if there is one
             var formula = result_cell.formula;
-            if(formula && formula.toUpperCase().includes("HYPERLINK")){
-               gamelink = formula.split("\"")[1];
+            
+            var gamelink;
+            if(!gamelink_id && formula && formula.toUpperCase().includes("HYPERLINK")){
+                //there is no gamelink, fill from the formula
+                gamelink = formula.split("\"")[1];
+            }else if(!result.gamelink_id){
+                //no gamelink and no formula
+                gamelink = '';
+            }else{
+                gamelink =  "http://en.lichess.org/" + result.gamelink_id;
```

Comment:
Constructing this URL is done in two places. not a big deal, but could get unified.